test/model/dataTypes/MemberVariable.test.ts

```diff
@@ -0,0 +1,271 @@
+import ListElement from "../../../src/model/modelInfo/ListElement";
+import SimpleElement from "../../../src/model/modelInfo/SimpleElement";
+import MemberVariable, {
+  RelationshipType,
+} from "../../../src/model/dataTypes/MemberVariable";
+import FilePath from "../../../src/model/dataTypes/FilePath";
+import SystemBoolean from "../../../src/model/dataTypes/system/SystemBoolean";
+import SystemString from "../../../src/model/dataTypes/system/SystemString";
+
+describe("MemberVariable", () => {
+  // Set up some test data
+  class BadElement extends Element {
+    constructor() {
+      super("bad");
+    }
+  }
+
+  let badElement: BadElement;
+  let simpleElement: SimpleElement;
+  let listElement: ListElement;
+  let choiceElement: ChoiceElement;
+  let fullMember: MemberVariable;
+
+  beforeEach(() => {
+    badElement = new BadElement();
+    simpleElement = new SimpleElement("person", "ns", "PrimitivePerson");
+    listElement = new ListElement("vehicle", "car", "Tesla");
+    const choice1: ChoiceType = {
+      typeName: "Chair",
+      namespace: "office",
+    };
+    const choice2: ChoiceType = {
+      typeName: "Bed",
+      namespace: "bedroom",
+    };
+
+    choiceElement = new ChoiceElement("position", [choice1, choice2]);
+
+    fullMember = new MemberVariable(
+      SystemBoolean,
+      "testVar",
+      true,
+      RelationshipType.HasMany,
+      false,
+      [SystemBoolean, SystemString]
+    );
+  });
+
+  // reusable assertion functions for above test data
+  function expectSimpleElementResult(member: MemberVariable): void {
+    expect(member.variableName).toBe("person");
+    expect(member.isArray).toBeFalse();
+    expect(member.dataType.namespace).toBe("ns");
+    expect(member.dataType.typeName).toBe("PrimitivePerson");
+    expect(member.dataType.path.value).toBe("/base/ns/PrimitivePerson");
+    expect(member.choiceTypes).toBeArrayOfSize(0);
+  }
+
+  function expectListElementResult(member: MemberVariable): void {
+    expect(member.variableName).toBe("vehicle");
+    expect(member.isArray).toBeTrue();
+    expect(member.dataType.namespace).toBe("car");
+    expect(member.dataType.typeName).toBe("Tesla");
+    expect(member.dataType.path.value).toBe("/base/car/Tesla");
+    expect(member.choiceTypes).toBeArrayOfSize(0);
+  }
+
+  function expectChoiceElementResult(member: MemberVariable): void {
+    expect(member.variableName).toBe("position");
+    expect(member.isArray).toBeFalse();
+    expect(member.dataType.namespace).toBe("FHIR");
+    expect(member.dataType.typeName).toBe("Type");
+    expect(member.dataType.path.value).toBe("/base/FHIR/Type");
+    expect(member.choiceTypes).toBeArrayOfSize(2);
+
+    expect(member.choiceTypes[0].namespace).toBe("office");
+    expect(member.choiceTypes[0].typeName).toBe("Chair");
+    expect(member.choiceTypes[0].path.value).toBe("/base/office/Chair");
+
+    expect(member.choiceTypes[1].namespace).toBe("bedroom");
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5966907)